src/coreclr/System.Private.CoreLib/src/System/Diagnostics/StackFrame.CoreCLR.cs

```diff
@@ -50,5 +53,23 @@ private void BuildStackFrame(int skipFrames, bool needFileInfo)
         }
 
         private static bool AppendStackFrameWithoutMethodBase(StringBuilder sb) => false;
+
+        [DllImport(RuntimeHelpers.QCall, EntryPoint = "StackFrame_GetMethodDescFromNativeIP")]
+        private static extern RuntimeMethodHandleInternal GetMethodDescFromNativeIP(IntPtr ip);
+
+        /// <summary>
+        /// Returns the method info instance for the managed code IP address.
+        /// </summary>
+        /// <param name="ip">code address</param>
+        /// <returns>MethodBase instance for the method or null if IP not found</returns>
+        internal static MethodBase? GetMethodFromNativeIP(IntPtr ip)
```

Comment:
We should probably add a comment right here saying that the implementation of this method has known race conditions for unloadable code + MethodBase isn't a good design choice for working with NativeAOT, therefore don't make it public in the future without considering those issues. Reduces the chance we forget the history when we need a public API later.

Reply:
Or even better - create issue that tracks adding the public API and link it from here

Reply:
I added a TODO describing most of this here: https://github.com/mikem8361/runtime/blob/b3a9a6086dcef769b82a5750489738254b6aee80/src/coreclr/vm/debugdebugger.cpp#L788

Reply:
Is that good enough?

Reply:
I did add a comment to the method's header.

Reply:
I do think having the message that you added right here in the header is valuable. When people convert a private API to public nothing enforces that they will look at implementation code in a separate file. The comment you added calls out the race condition but doesn't mention the design issue for using MethodBase in NativeAOT. As @jkotas suggested filing an issue and linking to that is an improvement over my original suggestion.